yt_dlp/extractor/kenh14.py

```diff
@@ -0,0 +1,150 @@
+        ],
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        attrs = extract_attributes(get_element_html_by_attribute('type', 'VideoStream', webpage) or '')
+        direct_url = attrs['data-vid']
+
+        metadata = self._download_json(
+            'https://api.kinghub.vn/video/api/v1/detailVideoByGet?FileName={}'.format(
+                remove_start(direct_url, 'kenh14cdn.com/')), video_id, fatal=False)
+
+        return {
+            'id': video_id,
+            'title': (
+                strip_or_none(metadata.get('title'))
+                or clean_html(self._og_search_title(webpage))
+                or clean_html(get_element_by_class('vdbw-title', webpage))),
+            'formats': [
+                {'url': f'https://{direct_url}', 'format_id': 'http'},
+                *self._extract_m3u8_formats(f'https://{direct_url}/master.m3u8', video_id, fatal=False),
+            ],
+            'duration': parse_duration(metadata.get('duration')),
+            'description': (
+                clean_html(self._og_search_description(webpage))
+                or clean_html(get_element_by_class('vdbw-sapo', webpage))),
+            'thumbnail': (self._og_search_thumbnail(webpage) or attrs.get('data-thumb')),
+            'uploader': strip_or_none(metadata.get('author')),
+            'timestamp': parse_iso8601(metadata.get('uploadtime'), delimiter=' '),
+            'view_count': int_or_none(metadata.get('views')),
+            'tags': traverse_obj(self._html_search_meta('keywords', webpage), (
+                {lambda x: x.split(';')}, lambda _, v: v, {str_or_none})),
+        }
+
+
+class Kenh14PlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://video\.kenh14\.vn/playlist/[\w-]+-(?P<id>[0-9]+)\.chn'
+    _TESTS = [{
+        'url': 'https://video.kenh14.vn/playlist/tran-tinh-naked-love-mua-2-71.chn',
+        'info_dict': {
+            'id': '71',
+            'title': 'Trần Tình (Naked love) mùa 2',
+            'description': 'md5:e9522339304956dea931722dd72eddb2',
+            'thumbnail': r're:^https?://kenh14cdn\.com/.*\.png$',
+        },
+        'playlist_count': 9,
+    }, {
+        'url': 'https://video.kenh14.vn/playlist/0-72.chn',
+        'info_dict': {
+            'id': '72',
+            'title': 'Lau Lại Đầu Từ',
+            'description': 'Cùng xem xưa và nay có gì khác biệt nhé!',
+            'thumbnail': r're:^https?://kenh14cdn\.com/.*\.png$',
+        },
+        'playlist_count': 6,
+    }]
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        webpage = self._download_webpage(url, playlist_id)
+
+        category_detail = get_element_by_class('category-detail', webpage)
+        embed_info = traverse_obj(
+            self._yield_json_ld(webpage, playlist_id),
+            (lambda _, v: v['name'] and v['alternateName']), get_all=False)
```

Comment:
```suggestion
            (lambda _, v: v['name'] and v['alternateName'], any)) or {}
```

Reply:
Hmm, the new code works but I don't know why? Some explanation?

Reply:
`any` does the same thing as `get_all=False` but inline, so you can also use it for dict traversal:
```py
traverse_obj({0: [0, 1], 1: [None 1]}, {
    'a': [0, ..., any],
    'b': [1, ..., filter, all],
})
```

See also the docstring, but the explanation is kinda bad. It sets branching to false and just takes the first item. Similarly `all` which sets branching to false and takes all items into a list.

PS: ideas for better developer documentation are welcome

Reply:
The example should be this:

```python
# -----------------------------------:
#                                    |
#                                    v
>>> traverse_obj({0: [0, 1], 1: [None, 1]}, {
        'a': [0, ..., any],
        'b': [1, ..., all],
    })

{'a': 0, 'b': [1]}

```

?

---

> ... It sets branching to false...

Hmmmm, the `traverse_obj` function becomes a bit fuzzier to me. 😄

> PS: ideas for better developer documentation are welcome

Okay. I've recorded it in my GitHub's [project board](https://github.com/pzhlkj6612?tab=projects).